### ISSUE TITLE
Is there a XOR equivalent?

### ISSUE THREAD

**Issue by fabienwarniez** [runtypes/runtypes#247]
Hi there, not really an issue, more of a question.

I am trying to validate that the payload contains either property `a` or property `b` but not both.

How can I achieve that elegantly?

Note: the case where both are present can either be a success or a failure, I would be happy with either.

Thank you in advance, and thanks for creating such an amazing library!

**Comment by Deco** [comment]
The standard TS typing of this is:
```ts
type T = { a: string; } | { b: string; }
```

One way to achieve this type in Runtypes is:
```ts
const T = Union(Record({ a: String, }), Record({ b: String}));
```

There are other ways, each with their own advantages and disadvantages.

**Comment by yuhr** [comment]
> The standard TS typing of this is:

No, that's not correct because it allows `const t: T = { a: 'foo', b: 'bar' };`. The OP wants to allow only either, not both.

I thought it could be achieved by this:

```ts
Union(Record({ a: Unknown, b: Never }), Record({ a: Never, b: Unknown }))
```

But it doesn't work. I believe this is a bug (`Never` property is not working expectedly), and will be fixed soon.

Using `Never` like above will become a pain if you have a ton of properties. Once "exact match" (#41, #169) has been introduced, you would be able to simply unionize exact `Record`s to do that.

**Comment by yuhr** [comment]
`Object({ ... }).exact()` modifier has been merged, so this will be  solved naturally in v7.
